dts/bindings/mtd/nxp,s32-qspi-device.yaml

```diff
@@ -20,3 +20,10 @@ properties:
       Memory alignment in bytes, used to calculate padding when performing
       unaligned accesses.
       If not provided, 1 byte alignment will be selected.
+
+  program-page-size:
+    type: int
+    required: true
+    description: |
+      The programming page size of the flash memory device,
+      as specified in the flash memory device datasheet.
```

Comment:
this property differs with `write-block-size`, this defines maximum of  programming  page buffer size configuration for hardware device flash. To avoid confusing I renamed this property to `max-program-buffer-size`.
As specified configuration in the flash memory device datasheet.
![image](https://github.com/user-attachments/assets/e27ed4f3-9c97-48bb-bd3f-3e1d0eb2a4f2)

Reply:
Where is the write block size set then, that is required for flash, and erase block size?

Reply:
write block size is defined by https://github.com/zephyrproject-rtos/zephyr/pull/78073/files#diff-aa5c7e912c23e4bbc1cabe0abb767637bcdaf3f9892233d71bd0841fdbec5c93R13
ease block size is defined by https://github.com/zephyrproject-rtos/zephyr/pull/78073/files#diff-626f2f951a3779218f24c5da9f599d4a6bde9f3fca7e6656cee9531ac1fc1277L46

Reply:
write block size being a define in code is useless and wrong, there is a dts property, likewise erase block size should be in dts, it's a hardware property (the page size is different, that is what size you want to erase, it is not the erase size)

Reply:
I see that `write-block-size` and `erase-block-size`, defined in https://github.com/zephyrproject-rtos/zephyr/blob/main/dts/bindings/mtd/soc-nv-flash.yaml which uses for internal flash.
I refer the other support external flash, I see most them not use these property.
eg. a supported driver that not use these. the write block size is fixed.
https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/flash/nrf_qspi_nor.c#L1172

Reply:
Hello @nordicjm, I’ve addressed all your comments. Could you please revisit the PR? I hope you happy with this and can approve my changes so we can merge it soon. Thanks.